### ISSUE TITLE
Why Uniform Average and Model Stock got almost the same acc in post-training setting?

### ISSUE THREAD

**Issue by Yikai-Liao** [naver-ai/model-stock#1]
![image](https://github.com/user-attachments/assets/8ebc0817-4a48-432f-b67c-0e7c96d9ab09)

With less than 5 finetuned checkpoints, how could `Uniform Average` perform almost the same as `Model Stock`, considering that `Model Stock` could achieve slightly better performance in previous experiements, where dozens of finetuned checkpoints were used for `Uniform Average`.

Does that means, more finetuned checkpoints does not lead to better performance for `Uniform Average` ?

![image](https://github.com/user-attachments/assets/f076d84e-969d-42a5-bee3-ec1e07981941)

**Comment by hellbell** [comment]
@Yikai-Liao 

Hi, thank you for asking questions about the model stock! 

> With less than 5 finetuned checkpoints, how could Uniform Average perform almost the same as Model Stock, considering that Model Stock could achieve slightly better performance in previous experiements, where dozens of finetuned checkpoints were used for Uniform Average.

In the post-training setup, the uniform average and model stock show similar `ImageNet` accuracy, but our main focus is obtaining robustness in the finetuning scenario. The model stock shows higher `distribution shifts` accuracies with closer distance from the center. 

> Does that means, more finetuned checkpoints does not lead to better performance for Uniform Average ?

For clarification, 
- `Uniform Average` in Table 7 is the uniform average of multiple checkpoints where each model was trained with different random seeds. 
- `Uniform Model Soups` in Table 2 is the average of multiple checkpoints where each model was trained with different hyper-parameters. (https://arxiv.org/abs/2203.05482)

Thus, the model stock (with post-training setup) performs better than the uniform average, especially in the out-of-distribution scenarios as in Table 7. In Table 2, model stock (with periodic merging) shows comparable performance against `Model Soups` with much smaller fine-tuning costs. 

I hope this helps you, and please let us know if you have further questions.
